lib/fetch/headers.js

```diff
@@ -91,64 +78,71 @@ function fill (headers, object) {
   }
 }
 
-// TODO: Composition over inheritence? Or helper methods?
-class HeadersList extends Array {
+class HeadersList {
+  constructor (init) {
+    if (init instanceof HeadersList) {
+      this[kHeadersMap] = new Map(init[kHeadersMap])
+      this[kSorted] = init[kSorted]
+      this[kHeadersSortedMap] = init[kHeadersSortedMap]
+    } else {
+      this[kHeadersMap] = new Map(init)
+      this[kSorted] = false
+      this[kHeadersSortedMap] = new Map()
```

Comment:
```suggestion
      this[kHeadersSortedMap] = null
```

If we have a sortedMap then it's sorted otherwise it's not... no need to have to vars here